src/shared/hostname-setup.c

```diff
@@ -206,7 +206,10 @@ int hostname_setup(bool really) {
                 if (enoent)
                         log_info("No hostname configured, using fallback hostname.");
 
-                hn = get_fallback_hostname();
+                hn = b = get_fallback_hostname();
```

Comment:
might like previous `b` here, no? or change `if (isempty(hn))` conditionalization to `if (hn)`?